forks/artifacts/Dockerfile.centos9

```diff
@@ -0,0 +1,24 @@
+FROM registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.18-openshift-4.12 AS oc
+ARG OS_GIT_VERSION="4.12.0"
+
+WORKDIR /go/src/github.com/openshift/oc
+COPY . .
+
+RUN yum install -y --setopt=skip_missing_names_on_install=False rpm-build createrepo bsdtar krb5-devel rsync
+RUN TMPDIR=$( mktemp -d ) RPMBUILD_DIR=_rpmbuild/ && mkdir -p ${RPMBUILD_DIR}/{BUILD,RPMS,SOURCES,SPECS,SRPMS} && \
+    tar -czf "${TMPDIR}/openshift-clients-0.0.1.tar.gz" --exclude='.git' \
+      --transform='s|^.|openshift-clients-0.0.1|' --exclude='openshift-clients-0.0.1.tar.gz' ./ && \
+    cp ./oc.spec "${RPMBUILD_DIR}/SPECS/" && \
+    cp "${TMPDIR}/openshift-clients-0.0.1.tar.gz" "${RPMBUILD_DIR}/SOURCES/" && \
+    rpmbuild -ba --nodeps --define "_topdir $( pwd )/${RPMBUILD_DIR}" \
+      ${RPMBUILD_DIR}/SPECS/oc.spec --define "os_git_vars OS_GIT_VERSION='${OS_GIT_VERSION}' \
+      OS_GIT_COMMIT='${OPENSHIFT_BUILD_COMMIT::8}' OS_GIT_MAJOR='' OS_GIT_MINOR=''  \
+      OS_GIT_TREE_STATE=''"
```

Comment:
The way this is done in Prow CI is just awful, let's do it properly here:
```suggestion
RUN yum install -y --setopt=skip_missing_names_on_install=False \
        rpm-build \
        createrepo \
        bsdtar \
        krb5-devel \
        rsync \
    && TMPDIR=$( mktemp -d ) \
    && RPMBUILD_DIR=_output/local/releases \
    && mkdir -p ${RPMBUILD_DIR}/{BUILD,SOURCES,SPECS,SRPMS} \
    && SOURCE_GIT_COMMIT=$(git rev-parse --short "HEAD^{commit}" 2>/dev/null) \
    && SOURCE_GIT_TAG=$(git describe --long --tags --abbrev=7 --match '*[0-9]*' | sed -E 's/openshift-clients-([0-9]+\.[0-9]+\.[0-9]+)-.*/\1/' || echo '0.0.0-unknown-${SOURCE_GIT_COMMIT}') \
    && SOURCE_GIT_TREE_STATE=$(( ( [ ! -d ".git/" ] || git diff --quiet ) && echo 'clean' ) || echo 'dirty') \
   #&& SOURCE_GIT_MAJOR= \
   #&& SOURCE_GIT_MINOR= \
  && RPM_VERSION=$(set -o pipefail && echo '${SOURCE_GIT_TAG}' | sed -E 's/v([0-9]+\.[0-9]+\.[0-9]+)-.*/\1/').$(shell date -u +'%Y%m%d%H%M').${SOURCE_GIT_COMMIT} \
  && tar -czf "${TMPDIR}/openshift-clients-${RPM_VERSION}.tar.gz" \
    --exclude='.git' \
    --transform='s|^.|openshift-clients-${RPM_VERSION}|' \
    --exclude='openshift-clients-${RPM_VERSION}.tar.gz' \
    ./ \
  && cp -ivh ./oc.spec "${RPMBUILD_DIR}/SPECS/" \
  && cp -ivh "${TMPDIR}/openshift-clients-${RPM_VERSION}.tar.gz" "${RPMBUILD_DIR}/SOURCES/" \
  # TODO: add keys and --sign
  && rpmbuild -ba --nodeps \
    --define "_topdir $(pwd)/${RPMBUILD_DIR}" \
    --define "_rpmdir $(pwd)/${RPMBUILD_DIR}/rpms" \
    --define 'version ${RPM_VERSION}' \
    --define "os_git_vars OS_GIT_VERSION='${SOURCE_GIT_TAG}' OS_GIT_COMMIT='${SOURCE_GIT_COMMIT}' OS_GIT_MAJOR='${SOURCE_GIT_MAJOR}' OS_GIT_MINOR='${SOURCE_GIT_MIJOR}' OS_GIT_TREE_STATE='${SOURCE_GIT_TREE_STATE}'" \
    ${RPMBUILD_DIR}/SPECS/oc.spec
```

Reply:
:+1: